### DISCUSSION TITLE
x-mask $money() input allows characters

### DISCUSSION THREAD

**Discussion by finesheer** [alpinejs/alpine#3302] [7. Bugs (archived)] [answered]
x-mask $money() input allows characters that are not numeric.

`<input x-mask:dynamic="$money($input)">
`

This is the example from the docs. If you visit  [#Money Inputs] (https://alpinejs.dev/plugins/mask#money-inputs) , you can reproduce.

<img width="769" alt="Screenshot 2022-11-09 at 6 54 30 PM" src="https://user-images.githubusercontent.com/32717047/200967062-e5f9a75e-e7c6-4347-a67f-dd55a06f2e6d.png">

**Comment by gaetan-hexadog** [comment]
Just saw it too.

If the first char is not a number then we can put any characters.
But once we put a number the previous non-number characters are deleted and we can't add characters other than numbres.

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423

**Comment by joshhanley** [comment, marked as answer]
This has been fixed in PR #3174 🙂
